src/spox/_var.py

```diff
@@ -24,28 +24,10 @@ def _not_impl(self, *_):
 
 class VarInfo:
     """
-    Abstraction for a single ONNX value - like a tensor - that can be passed around in Python code.
-
-    A ``VarInfo`` represents some output of an operator.
-    This operator is stored internally to allow reproducing the graph.
-
-    The ``type`` field is inferred and checked by operators.
-    It may be ``None`` if type inference failed, in which case it is unknown and should pass all type checks.
-    However, untyped ``VarInfo`` objects may not be used in some contexts.
-    Keep in mind that the types themselves may have some information missing.
-    For instance, tensors allow missing rank and shape information.
-
-    There is an implicit value propagation mechanism, powered by the ONNX reference implementation.
-    Values may be propagated if a ``VarInfo`` always has a known and constant value at runtime.
-    This is used for type & shape inference. For instance, Reshape to a constant shape can have the shape inferred.
+    Internal information about a ``Var``. Should be mainly inaccessible for most uses of ``spox``.
```

Comment:
"most"? Does `VarInfo` become part of the public interface?

Reply:
That is a good point, to be honest I was thinking about marking it as `_VarInfo` instead. I was really waiting to consolidate on the implementation to see if it had to pop-out somewhere.  

Now that I have gone through some of the downstream users of `spox`, I think it is safe to say we can mark it private.